tests/integration/security/file_mounted_certs/p2p_mtls_test.go

```diff
@@ -50,43 +51,58 @@ func TestClientToServiceTls(t *testing.T) {
 		Features("security.peer.file-mounted-certs").
 		Run(func(ctx framework.TestContext) {
 
-			echoClient, echoServer, serviceNamespace := setupEcho(t, ctx)
+			echoClients, echoServers, serviceNamespace := setupEcho(t, ctx)
 
 			bufDestinationRule := createObject(ctx, serviceNamespace.Name(), DestinationRuleConfigMutual)
 			defer ctx.Config().DeleteYAMLOrFail(ctx, serviceNamespace.Name(), bufDestinationRule)
 
 			bufPeerAuthentication := createObject(ctx, "istio-system", PeerAuthenticationConfig)
 			defer ctx.Config().DeleteYAMLOrFail(ctx, "istio-system", bufPeerAuthentication)
 
-			retry.UntilSuccessOrFail(t, func() error {
-				resp, err := echoClient.Call(echo.CallOptions{
-					Target:   echoServer,
-					PortName: "http",
-					Scheme:   scheme.HTTP,
-				})
+			callCount := 1
+			if ctx.Clusters().IsMulticluster() {
+				// so we can validate all clusters are hit
+				callCount = util.CallsPerCluster * len(ctx.Clusters())
+			}
+
+			for _, cluster := range ctx.Clusters() {
+				ctx.NewSubTest(fmt.Sprintf("From %s", cluster.Name())).Run(func(ctx framework.TestContext) {
+					echoClient := echoClients.GetOrFail(ctx, echo.InCluster(cluster))
+					retry.UntilSuccessOrFail(t, func() error {
+						resp, err := echoClient.Call(echo.CallOptions{
+							Target:   echoServers[0],
```

Comment:
doesn't `[0]` mean we only send to a single cluster? Should we be calling all instances?

Reply:
I think all instances of the same service will be hit? Referring to https://github.com/istio/istio/wiki/Writing-Good-Tests, 
```
func TestCrossClusterLoadBalancing(t *testing.T) {
  framework.NewTest(t).Run(func(ctx framework.TestContext) {
    // loop through the source services
    for _, a := a {
    a := a
    t.NewSubtest(a.Config().Cluster.Name()).Run(func(ctx framework.TestContext) {
      // we only need to target b[0]; all `b` services should be hit since they have the same name
      // set Count to something proportional to the number of possible targets to give load-balancing a chance to work
      res := a.CallOrFail(ctx, echo.CallOptions{Target: b[0], Count: 2*len(b)})
      // ensure 100% success
      res.CheckOKOrFail(ctx)
      // verify we reached all instances by using ParsedResponse
      clusterHits := map[string]int{}
      for _, r := range responses {
          clusterHits[r.Cluster]++
      }
      if len(hits) < targetCount {
        ctx.Fatal("did not hit all clsuters")
      }
    })
  })
}
```